Upgrade to rel/7.0.0

jeromelaban opened the pull request.

john-h-k committed:
Add NullableHelper

john-h-k committed:
Use more appropriate type in remap type

mrlacey committed:
Don't have the GridSplitterHandle show in UIA tree

For #3399

Sergio0694 committed:
Restructured the FindParent extensions

Sergio0694 committed:
Added FindChildOrSelf overloads

Sergio0694 committed:
Refactored enumerable parent/child visual tree extensions

Sergio0694 committed:
Updated leftover references in the codebase

Sergio0694 committed:
Fixed some more leftover references

Sergio0694 committed:
Fixed bug with reflection in TryGetContentControl

Sergio0694 committed:
Fixed a bug in TokenizingTextBox

Sergio0694 committed:
Fixed remaining unit tests

Sergio0694 committed:
Changed extensions namespaces in .NET packages

Sergio0694 committed:
Fixed naming of two XAML markup extensions

Sergio0694 committed:
Changed extensions namespaces in UWP packages

Sergio0694 committed:
Moved some single-file extensions

Sergio0694 committed:
Refactored the Mouse extensions class

Sergio0694 committed:
Refactored the TextBoxRegEx extensions

Sergio0694 committed:
Refactored the Visual/Logical tree extensions

Sergio0694 committed:
Refactored remaining TextBox extensions

Sergio0694 committed:
Refactored Matrix/Rect extensions

Sergio0694 committed:
Fixed visibility of some Media internal extensions

Sergio0694 committed:
Fixed some build errors in the sample app

Sergio0694 committed:
Renamed "uwpui" XML namespace to "ui"

Sergio0694 committed:
Fixed more leftover .Extensions references

Sergio0694 committed:
Fixed some build errors in the HighPerformance tests

Sergio0694 committed:
Fixed .NET Native .rd.xml directives in UWP test project

Sergio0694 committed:
Fixed a build error in the UI test app

Sergio0694 committed:
Fixed a leftover extension in a unit test

Sergio0694 committed:
Fixed a unit test failure due to incorrect extension matching

Sergio0694 committed:
Fixed some build errors in UWP unit tests

Sergio0694 committed:
Fixed an ArgumentNullException in TryGetContentControl

Sergio0694 committed:
Refactored the nullable extension

Sergio0694 committed:
Added NullableExtensions tests

Sergio0694 committed:
Fixed some leftover unit test build errors

ismaelestalayo committed:
updated InAppNotification style and template

Sergio0694 committed:
Minor API improvements

Tweaked TryGetContentControl and TryFindResource, and added a throwing version of FindResource

ismaelestalayo committed:
forgot to restore background

ismaelestalayo committed:
restored other bits

Sergio0694 committed:
Added optimized content paths for logical tree extensions

Sergio0694 committed:
Added new internal predicate types

Sergio0694 committed:
Performance improvements, removed code duplication

Sergio0694 committed:
Added some manual tail recursion unwrapping

Sergio0694 committed:
Fixed incorrect MS Docs link

Sergio0694 committed:
Improved exception for FindResource method

Sergio0694 committed:
Fixed some inconsistent internal namespaces

Sergio0694 committed:
Fixed some rebase issues

Sergio0694 committed:
Minor tweaks to some internal namespaaces

Sergio0694 committed:
Added FrameworkElement.TryFindResource overload

Sergio0694 committed:
Fixed a typo in XML docs

Co-authored-by: Michael Hawker MSFT (XAML Llama) <24302614+michael-hawker@users.noreply.github.com>

Sergio0694 committed:
Moved UserControl test paths down in priority

Commit:
[Visual|Logical]Tree extensions revamp (#3685)

<!-- 🚨 Please Do Not skip any instructions and information mentioned below as they are all required and essential to evaluate and test the PR. By fulfilling all the required information you will be able to reduce the volume of questions and most likely help merge the PR faster 🚨 -->

<!-- 📝 It is preferred if you keep the "☑️ Allow edits by maintainers" checked in the Pull Request Template as it increases collaboration with the Toolkit maintainers by permitting commits to your PR branch (only) created from your fork.  This can let us quickly make fixes for minor typos or forgotten StyleCop issues during review without needing to wait on you doing extra work. Let us help you help us! 🎉 -->


## Closes #3487 
<!-- Add the relevant issue number after the "#" mentioned above (for ex: Fixes #1234) which will automatically close the issue once the PR is merged. -->

<!-- Add a brief overview here of the feature/bug & fix. -->

## PR Type
What kind of change does this PR introduce?
<!-- Please uncomment one or more that apply to this PR. -->

 - Feature 
<!-- - Code style update (formatting) -->
<!-- - Refactoring (no functional changes, no api changes) -->
<!-- - Build or CI related changes -->
<!-- - Documentation content changes -->
<!-- - Sample app changes -->
<!-- - Other... Please describe: -->


## Overview
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
There are some inconsistencies in the visual tree extensions mentioned in the linked issue, and some missing features.
This PR applies the changes mentioned in https://github.com/windows-toolkit/WindowsCommunityToolkit/issues/3487#issuecomment-765028697.

## APIs breakdown

<details>
    <summary><b>VisualTree (click to expand):</b></summary>
<br/>

```csharp
namespace Microsoft.Toolkit.Uwp.UI.Extensions
{
    public static class VisualTree
    {
        static FrameworkElement? FindDescendant(this DependencyObject element, string name, StringComparison comparisonType = StringComparison.Ordinal);
        static T? FindDescendant<T>(this DependencyObject element) where T : notnull, DependencyObject;
        static DependencyObject? FindDescendant(this DependencyObject element, Type type);
        static T? FindDescendant<T>(this DependencyObject element, Func<T, bool> predicate) where T : notnull, DependencyObject;
        static T? FindDescendant<T, TState>(this DependencyObject element, TState state, Func<T, TState, bool> predicate) where T : notnull, DependencyObject;
        static FrameworkElement? FindDescendantOrSelf(this DependencyObject element, string name, StringComparison comparisonType = StringComparison.Ordinal);
        static T? FindDescendantOrSelf<T>(this DependencyObject element) where T : notnull, DependencyObject;
        static DependencyObject? FindDescendantOrSelf(this DependencyObject element, Type type);
        static T? FindDescendantOrSelf<T>(this DependencyObject element, Func<T, bool> predicate) where T : notnull, DependencyObject;
        static T? FindDescendantOrSelf<T, TState>(this DependencyObject element, TState state, Func<T, TState, bool> predicate) where T : notnull, DependencyObject;

        static IEnumerable<DependencyObject> FindDescendants(this DependencyObject element);

        static FrameworkElement? FindAscendant(this DependencyObject element, string name, StringComparison comparisonType = StringComparison.Ordinal);
        static T? FindAscendant<T>(this DependencyObject element) where T : notnull, DependencyObject;
        static DependencyObject? FindAscendant(this DependencyObject element, Type type);
        static T? FindAscendant<T>(this DependencyObject element, Func<T, bool> predicate) where T : notnull, DependencyObject;
        static T? FindAscendant<T, TState>(this DependencyObject element, TState state, Func<T, TState, bool> predicate) where T : notnull, DependencyObject;
        static FrameworkElement? FindAscendantOrSelf(this DependencyObject element, string name, StringComparison comparisonType = StringComparison.Ordinal);
        static T? FindAscendantOrSelf<T>(this DependencyObject element) where T : notnull, DependencyObject;
        static DependencyObject? FindAscendantOrSelf(this DependencyObject element, Type type);
        static T? FindAscendantOrSelf<T>(this DependencyObject element, Func<T, bool> predicate) where T : notnull, DependencyObject;
        static T? FindAscendantOrSelf<T, TState>(this DependencyObject element, TState state, Func<T, TState, bool> predicate) where T : notnull, DependencyObject;

        static IEnumerable<DependencyObject> FindAscendants(this DependencyObject element);
    }
}
```
</details>

<details>
    <summary><b>LogicalTree (click to expand):</b></summary>
<br/>

```csharp
namespace Microsoft.Toolkit.Uwp.UI.Extensions
{
    public static class LogicalTree
    {
        static FrameworkElement? FindChild(this FrameworkElement element, string name, StringComparison comparisonType = StringComparison.Ordinal);
        static T? FindChild<T>(this FrameworkElement element) where T : notnull, FrameworkElement;
        static FrameworkElement? FindChild(this FrameworkElement element, Type type);
        static T? FindChild<T>(this FrameworkElement element, Func<T, bool> predicate) where T : notnull, FrameworkElement;
        static T? FindChild<T, TState>(this FrameworkElement element, TState state, Func<T, TState, bool> predicate) where T : notnull, FrameworkElement;
        static FrameworkElement? FindChildOrSelf(this FrameworkElement element, string name, StringComparison comparisonType = StringComparison.Ordinal);
        static T? FindChildOrSelf<T>(this FrameworkElement element) where T : notnull, FrameworkElement;
        static FrameworkElement? FindChildOrSelf(this FrameworkElement element, Type type);
        static T? FindChildOrSelf<T>(this FrameworkElement element, Func<T, bool> predicate) where T : notnull, FrameworkElement;
        static T? FindChildOrSelf<T, TState>(this FrameworkElement element, TState state, Func<T, TState, bool> predicate) where T : notnull, FrameworkElement;

        static IEnumerable<FrameworkElement> FindChildren(this FrameworkElement element);

        static FrameworkElement? FindParent(this FrameworkElement element, string name, StringComparison comparisonType = StringComparison.Ordinal);
        static T? FindParent<T>(this FrameworkElement element) where T : notnull, FrameworkElement;
        static FrameworkElement? FindParent(this FrameworkElement element, Type type);
        static T? FindParent<T>(this FrameworkElement element, Func<T, bool> predicate) where T : notnull, FrameworkElement;
        static T? FindParent<T, TState>(this FrameworkElement element, TState state, Func<T, TState, bool> predicate) where T : notnull, FrameworkElement;
        static FrameworkElement? FindParentOrSelf(this FrameworkElement element, string name, StringComparison comparisonType = StringComparison.Ordinal);
        static T? FindParentOrSelf<T>(this FrameworkElement element) where T : notnull, FrameworkElement;
        static FrameworkElement? FindParentOrSelf(this FrameworkElement element, Type type);
        static T? FindParentOrSelf<T>(this FrameworkElement element, Func<T, bool> predicate) where T : notnull, FrameworkElement;
        static T? FindParentOrSelf<T, TState>(this FrameworkElement element, TState state, Func<T, TState, bool> predicate) where T : notnull, FrameworkElement;

        static IEnumerable<FrameworkElement> FindParents(this FrameworkElement element);
        
        static UIElement? TryGetContentControl(this FrameworkElement element);
        static object? TryFindResource(this FrameworkElement element, object resourceKey);
    }
}
```
</details>

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../readme.md#supported)
- [ ] Pull Request has been submitted to the documentation repository [instructions](..\contributing.md#docs). Link: <!-- docs PR link -->
- [ ] Sample in sample app has been added / updated (for bug fixes / features)
    - [ ] Icon has been created (if new sample) following the [Thumbnail Style Guide and templates](https://github.com/windows-toolkit/WindowsCommunityToolkit-design-assets)
- [ ] New major technical changes in the toolkit have or will be added to the [Wiki](https://github.com/windows-toolkit/WindowsCommunityToolkit/wiki) e.g. build changes, source generators, testing infrastructure, sample creation changes, etc...
- [ ] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [X] Header has been added to all new source files (run *build/UpdateHeaders.bat*)
- [ ] Contains **NO** breaking changes

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. 
     Please note that breaking changes are likely to be rejected within minor release cycles or held until major versions. -->


## Other information
Opening as draft, missing updated/revamped unit tests.

michael-hawker committed:
Merge branch 'master' into bugfix/observablevalidator-errors

Sergio0694 committed:
Fixed some build errors in the smoke tests

Commit:
Bug fixes to ObservableValidator (#3764)

## Fixes #3763
<!-- Add the relevant issue number after the "#" mentioned above (for ex: Fixes #1234) which will automatically close the issue once the PR is merged. -->

<!-- Add a brief overview here of the feature/bug & fix. -->

## PR Type
What kind of change does this PR introduce?
<!-- Please uncomment one or more that apply to this PR. -->

- Bugfix
<!-- - Feature -->
<!-- - Code style update (formatting) -->
<!-- - Refactoring (no functional changes, no api changes) -->
<!-- - Build or CI related changes -->
<!-- - Documentation content changes -->
<!-- - Sample app changes -->
<!-- - Other... Please describe: -->


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
- Incorrect loading of display names for validated properties
- Repeated error messages after validation

## What is the new behavior?
<!-- Describe how was this issue resolved or changed? -->
Fixed the errors above.

Opening as a draft to have the CI run, but I'm still investigating the second issue from the list above.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] Tested code with current [supported SDKs](../readme.md#supported)
- [ ] ~~Pull Request has been submitted to the documentation repository [instructions](..\contributing.md#docs). Link: <!-- docs PR link -->~~
- [ ] ~~Sample in sample app has been added / updated (for bug fixes / features)~~
    - [ ] ~~Icon has been created (if new sample) following the [Thumbnail Style Guide and templates](https://github.com/windows-toolkit/WindowsCommunityToolkit-design-assets)~~
- [X] New major technical changes in the toolkit have or will be added to the [Wiki](https://github.com/windows-toolkit/WindowsCommunityToolkit/wiki) e.g. build changes, source generators, testing infrastructure, sample creation changes, etc...
- [X] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [X] Header has been added to all new source files (run *build/UpdateHeaders.bat*)
- [X] Contains **NO** breaking changes

Sergio0694 committed:
Renamed StretchDirection enum type and property

azchohfi committed:
Changed to Cake.GitVersioning and added 'nbgv cloud' on smoke test job.

Sergio0694 committed:
Refactored the Surface Dial extensions for TextBox

ismaelestalayo committed:
smaller buttons and  text

ismaelestalayo committed:
fixed button size and position

michael-hawker committed:
Fix Sample Loading though there's a missing code-path for registration still

FYI @Sergio0694

michael-hawker committed:
Add new license/icon properties to our NuGet config

Based on @kyaa-dost's initial investigation

Keeping old and new formats as not all clients support the new formats in all scenarios yet.

Sergio0694 committed:
Added missing property change handler for Surface Dial

michael-hawker committed:
Update 30 to 32 for DPI scaling benefit

michael-hawker committed:
Switch to License Expression for Simplicity

Keep embedded license file though as well.

michael-hawker committed:
Update Microsoft.Toolkit.Uwp.Notifications/Microsoft.Toolkit.Uwp.Notifications.nuspec

michael-hawker committed:
Merge pull request #3778 (#3771) from michael-hawker/ismaelestalayo-inappnot-restyle

Update InAppNotification style to match WinUI InfoBar

michael-hawker committed:
Fix Surface Dial Sample Loading

Was trying to bind to non-DependencyProperties

michael-hawker committed:
Merge branch 'master' into nuget-update

michael-hawker committed:
Merge branch 'master' into versionFix

michael-hawker committed:
Merge pull request #3776 from windows-toolkit/versionFix

Changed to Cake.GitVersioning and added 'nbgv cloud' on smoke test job.

michael-hawker committed:
Fix Surface Dial Helpers to function as well as before

Add RotationResolutionInDegrees customization
Still some issues with Menu deregistration tracked in #3774

michael-hawker committed:
Merge branch 'master' into nuget-update

michael-hawker committed:
Merge pull request #3777 from michael-hawker/nuget-update

Update license and icon based on @kyaa-dost's investigation

michael-hawker committed:
Remove unneeded if which I believe was confusing compiler and causing a warning as we do check at top of function.

michael-hawker committed:
Merge branch 'master' into refactoring/extensions-namespace

Commit:
Refactor namespaces for extension types (#3743)

## Related to #3422 
<!-- Add the relevant issue number after the "#" mentioned above (for ex: Fixes #1234) which will automatically close the issue once the PR is merged. -->

<!-- Add a brief overview here of the feature/bug & fix. -->

## PR Type
What kind of change does this PR introduce?
<!-- Please uncomment one or more that apply to this PR. -->

<!-- - Bugfix -->
<!-- - Feature -->
<!-- - Code style update (formatting) -->
- Refactoring
<!-- - Build or CI related changes -->
<!-- - Documentation content changes -->
<!-- - Sample app changes -->
<!-- - Other... Please describe: -->


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
Quoting from https://github.com/windows-toolkit/WindowsCommunityToolkit/issues/3422#issuecomment-774484219, we have the current situation with respect to extension methods in the Toolkit:

> On a related note - lately I've been thinking a bit about the various extensions in the toolkit in particular.
@mrlacey already pointed this out in the first post in this issue, and I agree that that inconsistency is a bit weird, especially now that newer packages have started to just place them in their respective root namespaces. Basically we have the following situation:
>
> - `Microsoft.Toolkit.Extensions` (from `Microsoft.Toolkit`)❌
> - `Microsoft.Toolkit.Extensions` (from `Microsoft.Toolkit.Diagnostics`*) ❌
> - `Microsoft.Toolkit.HighPerformance.Extensions` ❌
> - `Microsoft.Toolkit.Mvvm.Messaging` ✅
> - `Microsoft.Toolkit.Uwp.Extensions` ❌
> - `Microsoft.Toolkit.Uwp.UI.Extensions` ❌
> - `Microsoft.Toolkit.Uwp.UI.Animations` ✅
> - `Microsoft.Toolkit.Uwp.UI.Media`** ✅

## What is the new behavior?
<!-- Describe how was this issue resolved or changed? -->
This PR moves all the extensions (including XAML markup extensions) to the root namespace of each package.
It also fixes some related issues like inconsistent naming, incorrect method locations, etc.

## Open questions
Some extension types now have a number of properties that are only relevant when grouped together, as they refer a specific feature. For instance, all the SurfaceDial properties in `TextBoxExtensions`, or the ones related to Regex. @michael-hawker mentioned the "option" pattern that the Graph Controls have been using, which would look something like this:

```xml
<TextBox>
  <ui:TextBoxExtentions.SurfaceDial>
    <ui:SurfaceDialOptions MinValue="0" MaxValue="10" EnableHapticFeedback="True" ... />
  </ui:TextBoxExtensions.SurfaceDial>
</TextBox>
```

Do we want to make this change as well while we're at it? Thoughts? 🙂

## Additional details

Haven't fully tested all the samples in the sample app just yet, have just focused on ensuring everything built fine for now.
Also, this PR is based on top of #3685, which needs to be merged before this one.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] Tested code with current [supported SDKs](../readme.md#supported)
- [ ] ~~Pull Request has been submitted to the documentation repository [instructions](..\contributing.md#docs). Link: <!-- docs PR link -->~~
- [ ] ~~Sample in sample app has been added / updated (for bug fixes / features)~~
    - [ ] ~~Icon has been created (if new sample) following the [Thumbnail Style Guide and templates](https://github.com/windows-toolkit/WindowsCommunityToolkit-design-assets)~~
- [X] New major technical changes in the toolkit have or will be added to the [Wiki](https://github.com/windows-toolkit/WindowsCommunityToolkit/wiki) e.g. build changes, source generators, testing infrastructure, sample creation changes, etc...
- [X] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [X] Header has been added to all new source files (run *build/UpdateHeaders.bat*)
- [ ] Contains **NO** breaking changes

michael-hawker committed:
Create 7.0.0-preview5 Release

Rosuavio committed:
Add space before comment

Rosuavio committed:
Remove unused _isParsing var

Rosuavio committed:
Split out multi class file

jamesmcroft committed:
Removed GetAutomationIdCore implementation from CarouselItem and BladeItem automation peers

jamesmcroft committed:
Merge branch 'master' into jamesmcroft/3506-carousel-automation

jamesmcroft committed:
Removed using directives which no longer exist

Rosuavio committed:
Include Microsoft.XamlIslands in UnitTest.XamlIslands.UWPApp
For some reason this fixes the build errors, it shouldint as its already a transitive dep.

azchohfi committed:
SampleApp fix. BitmapFileFormat moved assemblies.

michael-hawker committed:
Merge pull request #3507 from windows-toolkit/jamesmcroft/3506-carousel-automation

Added automation peer class for carousel and item

michael-hawker committed:
Fixes #3671 - No Case found scenario

michael-hawker committed:
Add basic SwitchPresenter sample to Sample App

shweaver-MSFT committed:
Adding dismissAll param to InAppNotifications Dismiss method

Rosuavio committed:
RangeSelector: Split out Properties file

Rosuavio committed:
RangeSelector: Split out Events file

shweaver-MSFT committed:
Restructuring use of custom notification

Rosuavio committed:
Made fuctions synchronous that were no longer using await

shweaver-MSFT committed:
Fixed bug in InAppNotification sample

Sergio0694 committed:
Removed System.ValueTuple package reference from Animations

Sergio0694 committed:
Moved numerics parsing extensions to StringExtensions type

Sergio0694 committed:
Optimizations to Vector2/3/4 and Quaternion parsing extensions

Sergio0694 committed:
Added internal builder support for delay behavior option

Sergio0694 committed:
Expanded delay behavior option to keyframe helpers

Sergio0694 committed:
Added DelayBehavior option to XAML animation types

Sergio0694 committed:
Added RepeatOption parameter to default animations

michael-hawker committed:
Merge branch 'master' into smoketest-winui

michael-hawker committed:
Merge pull request #3724 from michael-hawker/smoketest-winui

Add another Baseline for understanding WinUI overhead.

michael-hawker committed:
Remove complex property from SwitchPresenter

Fix name of TargetType DependencyProperty
Add better Enum string support

michael-hawker committed:
Merge branch 'master' into fix-xmalisland-test

michael-hawker committed:
Merge pull request #3783 from RosarioPulella/fix-xmalisland-test

[Bugfix] Include Microsoft.XamlIslands in UnitTest.XamlIslands.UWPApp

Sergio0694 committed:
Deprecated StringExtensions.AsFormat extension

jeromelaban committed:
merge from 7.0-pre5

michael-hawker committed:
Merge branch 'master' into bugfix/unnecessary-package

Sergio0694 committed:
Fixed a crash with the items list animations

shweaver-MSFT committed:
Fixed the bug in InAppNotifications with datatemplate

michael-hawker committed:
Merge pull request #3792 from Sergio0694/bugfix/unnecessary-package

Removed System.ValueTuple package reference from Animations

Kyaa-dost committed:
Merge branch 'master' into shweaver/inappnotifications

Commit:
Fixed a crash with the items list animations in Sample app (#3801)

<!-- 🚨 Please Do Not skip any instructions and information mentioned below as they are all required and essential to evaluate and test the PR. By fulfilling all the required information you will be able to reduce the volume of questions and most likely help merge the PR faster 🚨 -->

<!-- 📝 It is preferred if you keep the "☑️ Allow edits by maintainers" checked in the Pull Request Template as it increases collaboration with the Toolkit maintainers by permitting commits to your PR branch (only) created from your fork.  This can let us quickly make fixes for minor typos or forgotten StyleCop issues during review without needing to wait on you doing extra work. Let us help you help us! 🎉 -->


## Fixes #3701
<!-- Add the relevant issue number after the "#" mentioned above (for ex: Fixes #1234) which will automatically close the issue once the PR is merged. -->

<!-- Add a brief overview here of the feature/bug & fix. -->

## PR Type
What kind of change does this PR introduce?
<!-- Please uncomment one or more that apply to this PR. -->

- Bugfix (sample app)
<!-- - Feature -->
<!-- - Code style update (formatting) -->
<!-- - Refactoring (no functional changes, no api changes) -->
<!-- - Build or CI related changes -->
<!-- - Documentation content changes -->
<!-- - Sample app changes -->
<!-- - Other... Please describe: -->


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
There is a crash in the items animations when looking for samples. This was unrelated from the actual animation APIs, and the crash was happening because the parent of the `DropShadowPanel` for each item was actually `null` if the items were being swapped by a text change while one was animated, so the animation was being invoked on `null`, resulting in a crash. I've just added two checks to skip the animation out in case the whole list is being refreshed, as the item containers would all be discarded anyway.

## What is the new behavior?
<!-- Describe how was this issue resolved or changed? -->
The crash is fixed.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] Tested code with current [supported SDKs](../readme.md#supported)
- [ ] ~~Pull Request has been submitted to the documentation repository [instructions](..\contributing.md#docs). Link: <!-- docs PR link -->~~
- [ ] ~~Sample in sample app has been added / updated (for bug fixes / features)~~
    - [ ] ~~Icon has been created (if new sample) following the [Thumbnail Style Guide and templates](https://github.com/windows-toolkit/WindowsCommunityToolkit-design-assets)~~
- [X] New major technical changes in the toolkit have or will be added to the [Wiki](https://github.com/windows-toolkit/WindowsCommunityToolkit/wiki) e.g. build changes, source generators, testing infrastructure, sample creation changes, etc...
- [X] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [X] Header has been added to all new source files (run *build/UpdateHeaders.bat*)
- [X] Contains **NO** breaking changes

michael-hawker committed:
Merge pull request #3793 from Sergio0694/refactor/extensions

Refactored Uwp.UI visual extensions

shweaver-MSFT committed:
Merge branch 'master' into shweaver/inappnotifications

shweaver-MSFT committed:
Updated InAppNotification to use new DispatcherQueue

shweaver-MSFT committed:
Merge branch 'shweaver/inappnotifications' of https://github.com/windows-toolkit/WindowsCommunityToolkit into shweaver/inappnotifications

Sergio0694 committed:
Added internal notes with link to the WinUI issue

shweaver-MSFT committed:
Removed unnecessary async from InAppNotificaiton

michael-hawker committed:
Merge branch 'master' into deprecate/string-format

michael-hawker committed:
Merge pull request #3798 from Sergio0694/deprecate/string-format

Deprecated StringExtensions.AsFormat extension

Commit:
Animations package improvements (#3796)

## Follow up for #3639

<!-- Add a brief overview here of the feature/bug & fix. -->

## PR Type
What kind of change does this PR introduce?
<!-- Please uncomment one or more that apply to this PR. -->

- Feature
- Improvements
<!-- - Code style update (formatting) -->
<!-- - Refactoring (no functional changes, no api changes) -->
<!-- - Build or CI related changes -->
<!-- - Documentation content changes -->
<!-- - Sample app changes -->
<!-- - Other... Please describe: -->


## Changelog

This PR tracks new changes and tweaks for the `Microsoft.Toolkit.Uwp.UI.Animations` package before the 7.0 release.
Changes are done according to feedbacks and requests from developers trying out the Preview 5 release.
This PR contains the following changes:

- All `AnimationBuilder` and XAML animation types now use `AnimationDelayBehavior.SetValueBeforeDelay` by default
- New `DelayBehavior` property available for XAML animation types (ignored when an animation is on the XAML layer)
- New `delayBehavior` parameter available for the explicit `AnimationBuilder` keyframe animation APIs (the more advanced ones)
- New `RepeatOption` parameter now available for all `AnimationBuilder` APIs (default, keyframe, etc.)

Opening as a draft PR until we finalize the list of changes to add.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] Tested code with current [supported SDKs](../readme.md#supported)
- [ ] Pull Request has been submitted to the documentation repository [instructions](..\contributing.md#docs). Link: <!-- docs PR link -->
- [ ] Sample in sample app has been added / updated (for bug fixes / features)
    - [ ] Icon has been created (if new sample) following the [Thumbnail Style Guide and templates](https://github.com/windows-toolkit/WindowsCommunityToolkit-design-assets)
- [X] New major technical changes in the toolkit have or will be added to the [Wiki](https://github.com/windows-toolkit/WindowsCommunityToolkit/wiki) e.g. build changes, source generators, testing infrastructure, sample creation changes, etc...
- [ ] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [X] Header has been added to all new source files (run *build/UpdateHeaders.bat*)
- [ ] Contains **NO** breaking changes

Kyaa-dost committed:
Merge branch 'master' into shweaver/inappnotifications

Commit:
Fixed stack corruption bug in InAppNotification in sample (#3802)

<!-- 🚨 Please Do Not skip any instructions and information mentioned below as they are all required and essential to evaluate and test the PR. By fulfilling all the required information you will be able to reduce the volume of questions and most likely help merge the PR faster 🚨 -->

<!-- 📝 It is preferred if you keep the "☑️ Allow edits by maintainers" checked in the Pull Request Template as it increases collaboration with the Toolkit maintainers by permitting commits to your PR branch (only) created from your fork.  This can let us quickly make fixes for minor typos or forgotten StyleCop issues during review without needing to wait on you doing extra work. Let us help you help us! 🎉 -->


## Fixes #3391
<!-- Add the relevant issue number after the "#" mentioned above (for ex: Fixes #1234) which will automatically close the issue once the PR is merged. -->

<!-- Add a brief overview here of the feature/bug & fix. -->
In the InAppNotifications sample page, there is the option to display a notification with or without a custom DataTemplate. Mixing the usage of these buttons in the sample app with a StackMode other than "Replace", would cause cause the notifications to show double close buttons, or be out of alignment.

## PR Type
What kind of change does this PR introduce?
<!-- Please uncomment one or more that apply to this PR. -->

- Bugfix
<!-- - Feature -->
<!-- - Code style update (formatting) -->
<!-- - Refactoring (no functional changes, no api changes) -->
<!-- - Build or CI related changes -->
<!-- - Documentation content changes -->
<!-- - Sample app changes -->
<!-- - Other... Please describe: -->


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
Currently, the sample page for InAppNotifications shows the bug when interacting with the "... (with/without DataTemplate)" buttons. This is due to some odd wiring in the code behind. Though 3 notification objects were defined in XAML, only two were actually used. The third is configured to show the custom data template; But when it was time to show another notification it hot swaps templates with the custom notification, and attempts to show the custom content. Unfortunately this causes a bunch of issues because the need to display the close button and what content to show gets out of sync when we start to close the stack of notifications.

## What is the new behavior?
<!-- Describe how was this issue resolved or changed? -->
In the update sample page, I've started using the third notification object. It maintains a separate stack because it is a separate notification, but so does the VS styled notification. The advise to developers going forward is to pick a model, not mix and match. Hot-swapping templates is not supported.

To support the flow of the sample, I added an option to pass an argument to the Dismiss function.

`InAppNotification.Dismiss(bool dismissAll = false)`

Passing `true` will dismiss all of the notifications in the stack. I read that there was a comment in the sample code-behind saying, "Do not dismiss all in production", but I disagree. Sometimes you really need to dismiss them all, such as during major app navigations (sign in/out). If a whole stack of notifications becomes suddenly irrelevant we should be able to dismiss them all. It's useful in the sample as well, for when we switch between the three notifications.
 
Lastly, in this process I uncovered a bug that causes the notification not to render when reusing the same DataTemplate. Here is the snippet from InAppNotification.cs
```
// From InAppNotifcation.UpdateContent(NotificationOptions notificationOptions)

    case DataTemplate dataTemplate:
        // Without this check, the dataTemplate will fail to render.
        // Why? Setting the ContentTemplate causes the control to re-evaluate it's Content value.
        // When we set the ContentTemplate to the same instance of itself, we aren't actually changing the value.
        // This means that the Content value won't be re-evaluated and stay null, causing the render to fail.
        if (_contentProvider.ContentTemplate != dataTemplate)
        {
            _contentProvider.ContentTemplate = dataTemplate;
            _contentProvider.Content = null;
        }

        break;
```

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../readme.md#supported)
- [ ] Pull Request has been submitted to the documentation repository [instructions](..\contributing.md#docs). Link: <!-- docs PR link -->
- [x] Sample in sample app has been added / updated (for bug fixes / features)
    - [ ] Icon has been created (if new sample) following the [Thumbnail Style Guide and templates](https://github.com/windows-toolkit/WindowsCommunityToolkit-design-assets)
- [ ] New major technical changes in the toolkit have or will be added to the [Wiki](https://github.com/windows-toolkit/WindowsCommunityToolkit/wiki) e.g. build changes, source generators, testing infrastructure, sample creation changes, etc...
- [ ] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [x] Header has been added to all new source files (run *build/UpdateHeaders.bat*)
- [x] Contains **NO** breaking changes

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. 
     Please note that breaking changes are likely to be rejected within minor release cycles or held until major versions. -->


## Other information
Todo:
- [ ] Document the new dismissAll flag.

michael-hawker committed:
Merge branch 'master' into master

Commit:
Add NullableHelper (#3389)

cc @Sergio0694 

Introduce a `NullableHelper` type with `GetValueOrDefaultRef` which returns a `ref T` which points to a value that is equivalent to `GetValueOrDefault`

TODO add tests

michael-hawker committed:
Merge branch 'master' into fix-local-release-build

michael-hawker committed:
Merge pull request #3761 from RosarioPulella/fix-local-release-build

Fix Visual studio release build

michael-hawker committed:
Remove duplicate style from WrapPanel sample

michael-hawker committed:
Clean-up sample and better separate the two examples

michael-hawker committed:
Remove SwitchPresenter logic that would listen to changes in the collection or case values

Can't think of scenarios where this was actually useful and added a lot of complexity.
Removing it for now for the core value the control provides and simplifies the structure.
Samples from Sample App continue to work as expected.

michael-hawker committed:
🦙 Add more 🦙

Sergio0694 committed:
Added AnimationBuilder C# sample code

Commit:
Add basic SwitchPresenter Sample (#3789)

## Also Fixes #3671, and related to #1908
Adds a basic sample for SwitchPresenter to the sample app. Would like to do a more advanced one still though. This also fixes a small bug in the case selection logic.

_Any suggestions for sample scenarios where we could switch off a strong type like an enum or some other type would be appreciated._ 🙂

## PR Type
What kind of change does this PR introduce?
<!-- Please uncomment one or more that apply to this PR. -->

- Bugfix
<!-- - Feature -->
<!-- - Code style update (formatting) -->
<!-- - Refactoring (no functional changes, no api changes) -->
<!-- - Build or CI related changes -->
<!-- - Documentation content changes -->
- Sample app changes
<!-- - Other... Please describe: -->

## What is the current behavior?
Would crash if no default case provided and no matching case available.

(And no sample.)

## What is the new behavior?
Control goes back to presenting nothing if no match or default is available.

+ Sample!

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../readme.md#supported)
- [ ] Pull Request has been submitted to the documentation repository [instructions](..\contributing.md#docs). Link: <!-- docs PR link -->
- [ ] Sample in sample app has been added / updated (for bug fixes / features)
    - [ ] Icon has been created (if new sample) following the [Thumbnail Style Guide and templates](https://github.com/windows-toolkit/WindowsCommunityToolkit-design-assets)
- [ ] New major technical changes in the toolkit have or will be added to the [Wiki](https://github.com/windows-toolkit/WindowsCommunityToolkit/wiki) e.g. build changes, source generators, testing infrastructure, sample creation changes, etc...
- [ ] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Header has been added to all new source files (run *build/UpdateHeaders.bat*)
- [ ] Contains **NO** breaking changes

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. 
     Please note that breaking changes are likely to be rejected within minor release cycles or held until major versions. -->

## TO DO

- [x] Add an Advanced scenario to the page using TargetType
- [x] Address #3784 as well? (Think un-needed for now?) App can always do this if needed.
- [x] Remove listening for Child Value binding changes as complicates the control.

jeromelaban committed:
chore: Adjust for preview5 merge

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/unoplatform/Uno.WindowsCommunityToolkit?pullRequest=127) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/unoplatform/Uno.WindowsCommunityToolkit?pullRequest=127) before we can accept your contribution.<br/>**2** out of **6** committers have signed the CLA.<br/><br/>:white_check_mark: jeromelaban<br/>:white_check_mark: michael-hawker<br/>:x: Sergio0694<br/>:x: azchohfi<br/>:x: RosarioPulella<br/>:x: msftbot[bot]<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/unoplatform/Uno.WindowsCommunityToolkit?pullRequest=127) it.</sub>

michael-hawker committed:
Fixes #3629 TileControl crashes when moving monitors

The LoadCompleted event was never unregistering and firing twice when reloaded across monitors causing an issue with the TaskCompletionSource.

jeromelaban committed:
ci: Adjust for .NET 5

jeromelaban committed:
chore: build fixes

jeromelaban committed:
ci: Adjust linux build

jeromelaban committed:
ci: adjust TF override

jeromelaban committed:
ci: Adjust package reference

jeromelaban committed:
ci: adjust mathex

jeromelaban committed:
ci: Adjust warning

jeromelaban committed:
ci: other syntax changes

jeromelaban commented:
/azp run

Sergio0694 committed:
Renamed TextBoxExtensions.PlaceHolder property

Sergio0694 committed:
Updated TextBoxExtensions sample pages

Sergio0694 committed:
Renamed mask placeholder property

michael-hawker committed:
Add ContentPresenter to Logical Tree helpers

Helps get down through SwitchPresenter. (Up doesn't work as Parent is not being set, WinUI bug?)

michael-hawker committed:
Merge pull request #3809 from Sergio0694/sample/explicit-animations

Added AnimationBuilder C# sample code

yoshiask committed:
Update master (7.0 Preview 5)

Update master

yoshiask committed:
Added conditional namespace for ShouldConstrainToRootBounds

michael-hawker committed:
Update RangeMin and RangeMax to RangeStart and RangeEnd

These properties and their descriptions were confusing (and abbreviated) which made it difficult to understand how they differed from the Minimum/Maximum value properties.
This clearly now separates which properties are being selected by the user. Their comments have been updated as well.
It also aligns with the new Range struct field names in .NET: Start/End.

original author: @michael-hawker

Commit:
Changed TextBoxExtensions.PlaceHolder property name (#3810)

## PR Type
What kind of change does this PR introduce?
<!-- Please uncomment one or more that apply to this PR. -->

- Refactoring with API changes
<!-- - Bugfix -->
<!-- - Feature -->
<!-- - Code style update (formatting) -->
<!-- - Refactoring (no functional changes, no api changes) -->
<!-- - Build or CI related changes -->
<!-- - Documentation content changes -->
<!-- - Sample app changes -->
<!-- - Other... Please describe: -->


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
The mask placeholder property is called `PlaceHolder`.

## What is the new behavior?
<!-- Describe how was this issue resolved or changed? -->
It's now called `MaskPlaceholder` instead.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] Tested code with current [supported SDKs](../readme.md#supported)
- [X] Pull Request has been submitted to the documentation repository [instructions](..\contributing.md#docs). Link: <!-- docs PR link -->
- [X] Sample in sample app has been added / updated (for bug fixes / features)
    - [X] Icon has been created (if new sample) following the [Thumbnail Style Guide and templates](https://github.com/windows-toolkit/WindowsCommunityToolkit-design-assets)
- [X] New major technical changes in the toolkit have or will be added to the [Wiki](https://github.com/windows-toolkit/WindowsCommunityToolkit/wiki) e.g. build changes, source generators, testing infrastructure, sample creation changes, etc...
- [X] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [X] Header has been added to all new source files (run *build/UpdateHeaders.bat*)
- [X] Contains **NO** breaking changes

jeromelaban committed:
chore: Additional formatting updates

jeromelaban committed:
ci: always publish artifacts

jeromelaban committed:
chore: Adjust path

jeromelaban committed:
chore: Additional update

Sergio0694 committed:
Added removal for Storyboard completion handler

Sergio0694 committed:
Fixed unit tests for Point/Rect/Size extensions

Sergio0694 committed:
Merge pull request #3816 from Sergio0694/bugfix/storyboard-event-unsubscribe

Added removal for Storyboard completion handler

Commit:
Fixed unit tests for Point/Rect/Size extensions (#3817)

## PR Type
What kind of change does this PR introduce?
<!-- Please uncomment one or more that apply to this PR. -->

- Unit test fixes
<!-- - Bugfix -->
<!-- - Feature -->
<!-- - Code style update (formatting) -->
<!-- - Refactoring (no functional changes, no api changes) -->
<!-- - Build or CI related changes -->
<!-- - Documentation content changes -->
<!-- - Sample app changes -->
<!-- - Other... Please describe: -->


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
The tests for `Point`/`Rect`/`Size` extensions are skipped, and some are just incorrect anyway and fail even when run.

## What is the new behavior?
<!-- Describe how was this issue resolved or changed? -->
Fixed those tests to run correctly.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] Tested code with current [supported SDKs](../readme.md#supported)
- [ ] ~~Pull Request has been submitted to the documentation repository [instructions](..\contributing.md#docs). Link: <!-- docs PR link -->~~
- [ ] ~~Sample in sample app has been added / updated (for bug fixes / features)~~
    - [ ] ~~Icon has been created (if new sample) following the [Thumbnail Style Guide and templates](https://github.com/windows-toolkit/WindowsCommunityToolkit-design-assets)~~
- [X] New major technical changes in the toolkit have or will be added to the [Wiki](https://github.com/windows-toolkit/WindowsCommunityToolkit/wiki) e.g. build changes, source generators, testing infrastructure, sample creation changes, etc...
- [X] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [X] Header has been added to all new source files (run *build/UpdateHeaders.bat*)
- [X] Contains **NO** breaking changes

Sergio0694 committed:
Merge branch 'master' into fix-tilecontrol

michael-hawker committed:
Merge branch 'master' into fix/tabbedcommandbar_SCTRB

michael-hawker committed:
Merge pull request #3813 from yoshiask/fix/tabbedcommandbar_SCTRB

#3779 Fix TabbedCommandBar MinVersion

michael-hawker committed:
Update Microsoft.Toolkit.Uwp.UI.Controls.Core/TileControl/TileControl.cs

Co-authored-by: Sergio Pedri <sergio0694@live.com>

marcelwgn committed:
Switch to fast pass rendering

marcelwgn committed:
Switch to italic

michael-hawker committed:
Merge branch 'master' into fix-tilecontrol

michael-hawker committed:
Merge pull request #3808 from michael-hawker/fix-tilecontrol

Fixes #3629 TileControl crashes when moving monitors

michael-hawker committed:
Update XamlIslandsTest_StringExtensions to fix from Extensions namespace change

Related to #3743

Rosuavio committed:
Reorg RangeSelector code
into file based on type of input

Rosuavio committed:
Move prop changed handlers to prop file

Rosuavio committed:
Fallow the proper 'On___' pattern

michael-hawker committed:
Merge pull request #3821 from windows-toolkit/michael-hawker-patch-1

Update XamlIslandsTest_StringExtensions to fix from Extensions namesp…

Kyaa-dost committed:
Merge branch 'master' into dev/stringextensions-a11y

Rosuavio committed:
Make On__ events methods virtual

marcelwgn committed:
Fix issue with bold not being applied

Kyaa-dost committed:
Merge branch 'master' into dev/fast-pass-rendering

Rosuavio committed:
Moved static mathod above non static methods.

Rosuavio committed:
Remove blank lines between doc comment and method

Rosuavio committed:
Remove un-needed usings

Sergio0694 committed:
Skipped some redundant property accesses

Sergio0694 committed:
Merge branch 'master' into fix-switchpresenter-logicaltree

michael-hawker committed:
Merge pull request #3811 from michael-hawker/fix-switchpresenter-logicaltree

Add ContentPresenter to Logical Tree helpers

michael-hawker committed:
Merge branch 'master' into refactor/RangeSelector-SplitFile

michael-hawker committed:
Merge pull request #3822 from RosarioPulella/refactor/RangeSelector-SplitFile

[Refactor] Split up RangeSelector.cs file

michael-hawker committed:
Merge branch 'master' into trigger-enum-parse

michael-hawker committed:
Merge pull request #3678 from huoyaoyuan/trigger-enum-parse

Refine enum conversion logic in IsEqualStateTrigger

michael-hawker committed:
Merge branch 'master' into dev/stringextensions-a11y

michael-hawker committed:
Merge pull request #3820 from chingucoding/dev/stringextensions-a11y

StringExtensions: Switch to italic instead of blue, fix bug with bold effect not being applied

michael-hawker committed:
Merge branch 'master' into dev/fast-pass-rendering

michael-hawker committed:
Merge pull request #3819 from chingucoding/dev/fast-pass-rendering

Switch to fast path rendering

michael-hawker committed:
Merge pull request #3823 from michael-hawker/bring-gridsplitter-fix

Don't have the GridSplitterHandle show in UIA tree

jeromelaban committed:
chore: Update to uno latest

jeromelaban committed:
ci: update uno

jeromelaban committed:
ci: Update uno

michael-hawker committed:
Simplify Range Selector sample

michael-hawker committed:
Fix issue with ColorPicker not working with SwitchPresenter changes

Since optimizing SwitchPresenter, the delayed loading being in another control's template was causing an issue where the cases weren't available to evaluate when SwitchPresenter was ready. Since we no longer listened to all downstream changes in SwitchPresenter this meant we never evaluated once the parent control was loaded. Added a couple of cases to ensure that once we're fully loaded we can evaluate again to ensure we pick a case. This also covers a case where if somehow the entire collection gets swapped out later.

Added some extra checks ahead of the evaluate cases to break out early in case we don't need to update the Visual Tree.

michael-hawker committed:
Add 7.0 features to landing page

Fix issue with version history display on About page
Add Wiki Link
Add MVVM Toolkit entry to sample for docs
Clean-up ColorPickerPage sample

michael-hawker committed:
Apply XAML Styler changes

michael-hawker committed:
Fix Home Button being broken

Remove Custom NavigationView template, it should support everything we need now
Remove workaround for WinUI 2.4 bug
Add new workaround for another issue...

michael-hawker committed:
Update Sample App to properly open links found in docs now

Have docs in release mode look at live branch of last shipped docs.

michael-hawker committed:
Ensure code url goes to the same version as the package in release mode.

jeromelaban committed:
chore: Update to uno latest

jeromelaban committed:
ci: Adjust linux build parameters

jeromelaban committed:
fix: Adjust casing for resw files

jeromelaban committed:
fix: Disable roslyn generators for linux build

jeromelaban committed:
chore: Override uno version for all targets for sample app

jeromelaban committed:
chore: move version override

michael-hawker committed:
Add-in Graph Control docs

Can't add examples back as we haven't rebuilt the Graph Controls library yet.

azchohfi committed:
Moved a few things to use the DispatcherQueue.

Rosuavio committed:
A bunch of safe url fixes for the sample pages

Rosuavio committed:
More safe url fixes

Sergio0694 committed:
Fixed animation samples links and icons

Rosuavio committed:
Maybe good urls for VisualEffectFactory

Rosuavio committed:
Maybe good url for GazeTracking sample

Rosuavio committed:
Mark broken urls

Rosuavio committed:
Mark missing docs

jeromelaban committed:
ci: Adjust for net5 path

azchohfi committed:
Changed all DispatcherTimer to DispatcherQueueTimer.

Rosuavio committed:
Add description for Controls meta package

Sergio0694 committed:
Fixed outdated docs links

Rosuavio committed:
Update Microsoft.Toolkit.Uwp.UI.Controls/Microsoft.Toolkit.Uwp.UI.Controls.csproj

Co-authored-by: Michael Hawker MSFT (XAML Llama) <24302614+michael-hawker@users.noreply.github.com>

Rosuavio committed:
Fix & -> &amp;

Rosuavio committed:
Added Controsl readme file

Rosuavio committed:
Add winui 2.5 dep to all controls packages

Rosuavio committed:
EnumValuesExtnetion sample fix DocUrl

Rosuavio committed:
Fix sampel doc link AutoFocusBehavior + FocusBehavior

Rosuavio committed:
Add include for readme.txt file

michael-hawker committed:
Merge pull request #2 from RosarioPulella/sample-app-updates-url-fixes

Sample app updates url fixes

michael-hawker committed:
Merge remote-tracking branch 'sergio/sample/animation-apis' into sample-app-updates

michael-hawker committed:
Fix Fade doc link and extra comma

michael-hawker committed:
Update app version number to 7.0.0

michael-hawker committed:
Update Microsoft.Toolkit.Uwp.UI.Controls/readme.txt

michael-hawker committed:
Merge pull request #3830 from windows-toolkit/dispatcherQueue

Using DispatcherQueue instead of CoreDispatcher

michael-hawker committed:
Merge branch 'master' into sample-app-updates

michael-hawker committed:
Merge branch 'master' into controls-package-description

michael-hawker committed:
Merge pull request #3829 from RosarioPulella/controls-package-description

Add description for Controls meta package

azchohfi committed:
Update Microsoft.Toolkit.Uwp.SampleApp/Models/Sample.cs

Co-authored-by: Michael Hawker MSFT (XAML Llama) <24302614+michael-hawker@users.noreply.github.com>

azchohfi committed:
Merge branch 'master' into sampleAppMediaFix

michael-hawker committed:
Merge pull request #3785 from windows-toolkit/sampleAppMediaFix

SampleApp fix. BitmapFileFormat moved assemblies.

michael-hawker committed:
Merge branch 'master' into sample-app-updates

Commit:
Sample app updates for 7.0 (#3824)

## Fixes lots
Smorgasbord of changes found or related to updating the Sample App for 7.0 shipping.

- Fixes an issue with SwitchPresenter causing ColorPicker to break (Loading order for evaluating cases)
- Fixes an issue on About page displaying preview version notes
- Updates the Landing Page Links
- Adds MVVM Toolkit Docs
- Adds Wiki Link
- Cleans up some XAML only samples
- Applies XAML Styler

## TODO

- [x] Fix Home Button not working
- [x] Fix new short URLs for docs not having https://docs.microsoft.com/ to add pre-cursor
- [x] Should pin github urls to v7.0.0 in release mode, kind of like what we do for docs???
- [x] Update rest of doc links
  - [x] Update paths in CodeUrl
  - [x] Update paths to docs
  - [x] Check animation sample links
- [x] Add Animation's sample Image
- [x] Graph Controls Docs at least (samples on ice still)
- [ ] Build sample app in release mode for release

michael-hawker committed:
Update version.json for 7.0.0

jeromelaban committed:
Merge rel/7.0.0

jeromelaban committed:
ci: Adjust after merge

jeromelaban committed:
ci: adjust converter

jeromelaban committed:
chore: Adjust missing enum member

jeromelaban committed:
chore: adjust removed files
